### PR TITLE
[PCM-1510] pass jid resource to stanza

### PR DESCRIPTION
**Pull request by jensengar**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -20,10 +20,11 @@ let extensions = {
   webrtcSessions: WebrtcExtension
 };
 
-function stanzaioOptions (config) {
+function stanzaioOptions (config: ClientOptions & { channelId: string }): AgentConfig {
   let wsHost = config.host.replace(/\/$/, '');
-  let stanzaOptions = {
+  let stanzaOptions: AgentConfig = {
     jid: config.jid,
+    resource: config.jidResource,
```
Comment (djhouseknecht) [inline src/client.ts:27]: The fact that this is already spec'ed out is awesome. Great find, John Hjelle.

File: src/client.ts [hunk 2]
```diff
@@ -80,6 +81,7 @@ export interface ClientOptions {
   authToken?: string;
   jwt?: string;
   jid?: string;
+  jidResource?: string;
```
Comment (djhouseknecht) [inline src/client.ts:84]: Would it be too much to add code comments on these options? I find that really helpful for the code hinting and if I need to `cmd + click` the definition to take me to the `*.d.ts` file. 

Also wouldn't hurt to type out the `config` for the `Client` class.

**Review by djhouseknecht — CHANGES_REQUESTED** [review]
Two overal comments: 
1. I think the CHANGELOG needs to be updated with these changes
2. Does the version need to be bumped (since we don't get that for free anymore)?

File: src/client.ts (hunk 2 above)
Comment (jensengar) [reply src/client.ts:84]: Someday we'll have to change the doc to match whatever you are doing with the sdk.

**Review by djhouseknecht — APPROVED** [review]

**Pull request merged**